doc.js

```diff
@@ -394,6 +394,26 @@ define('doc', ['event'], function(event) {
 				} else {
 					return;
 				}
+			},
+
+			'insertBefore' : function(elements) {
+				if(typeof elements == 'string'){
+					elements = query(search(document, elements));
+				}
+
+				elements.each(function(el) {
+					el.insertAdjacentHTML('beforebegin', this.els[0].outerHTML);
```

Comment:
If the element has no parent this function throws an error. Suppose you wish to create the elements before inserting into the DOM tree. These elements will have no parent. You could work around it by inserting an element as a parent of `el` if `el` has no parent and remove it afterwards in order to make this work.

@mottam @erikatakahara